pkg/kubelet/winstats/network_stats.go

```diff
@@ -0,0 +1,312 @@
+		return nil, err
+	}
+
+	bytesSentPerSecondData, err := n.bytesSentPerSecondCounter.getDataList()
+	if err != nil {
+		klog.Errorf("Unable to get bytesSentPerSecond perf counter data; err: %v", err)
+		return nil, err
+	}
+
+	packetsReceivedDiscardedData, err := n.packetsReceivedDiscardedCounter.getDataList()
+	if err != nil {
+		klog.Errorf("Unable to get packetsReceivedDiscarded perf counter data; err: %v", err)
+		return nil, err
+	}
+
+	packetsReceivedErrorsData, err := n.packetsReceivedErrorsCounter.getDataList()
+	if err != nil {
+		klog.Errorf("Unable to get packetsReceivedErrors perf counter data; err: %v", err)
+		return nil, err
+	}
+
+	packetsOutboundDiscardedData, err := n.packetsOutboundDiscardedCounter.getDataList()
+	if err != nil {
+		klog.Errorf("Unable to get packetsOutboundDiscarded perf counter data; err: %v", err)
+		return nil, err
+	}
+
+	packetsOutboundErrorsData, err := n.packetsOutboundErrorsCounter.getDataList()
+	if err != nil {
+		klog.Errorf("Unable to get packetsOutboundErrors perf counter data; err: %v", err)
+		return nil, err
+	}
+
+	n.mu.Lock()
+	defer n.mu.Unlock()
+	n.mergeCollectedData(
+		packetsReceivedPerSecondData,
+		packetsSentPerSecondData,
+		bytesReceivedPerSecondData,
+		bytesSentPerSecondData,
+		packetsReceivedDiscardedData,
+		packetsReceivedErrorsData,
+		packetsOutboundDiscardedData,
+		packetsOutboundErrorsData,
+	)
+	return n.listInterfaceStats(), nil
+}
+
+// mergeCollectedData merges the collected data into cache. It should be invoked under lock protected.
+func (n *networkCounter) mergeCollectedData(packetsReceivedPerSecondData,
+	packetsSentPerSecondData,
+	bytesReceivedPerSecondData,
+	bytesSentPerSecondData,
+	packetsReceivedDiscardedData,
+	packetsReceivedErrorsData,
+	packetsOutboundDiscardedData,
+	packetsOutboundErrorsData map[string]uint64) {
+	adapters := sets.NewString()
+
+	// merge the collected data and list of adapters.
+	adapters.Insert(n.mergePacketsReceivedPerSecondData(packetsReceivedPerSecondData)...)
+	adapters.Insert(n.mergePacketsSentPerSecondData(packetsSentPerSecondData)...)
+	adapters.Insert(n.mergeBytesReceivedPerSecondData(bytesReceivedPerSecondData)...)
+	adapters.Insert(n.mergeBytesSentPerSecondData(bytesSentPerSecondData)...)
+	adapters.Insert(n.mergePacketsReceivedDiscardedData(packetsReceivedDiscardedData)...)
+	adapters.Insert(n.mergePacketsReceivedErrorsData(packetsReceivedErrorsData)...)
+	adapters.Insert(n.mergePacketsOutboundDiscardedData(packetsOutboundDiscardedData)...)
+	adapters.Insert(n.mergePacketsOutboundErrorsData(packetsOutboundErrorsData)...)
+
+	// delete the cache for non-existing adapters.
+	for adapter := range n.adapterStats {
+		if !adapters.Has(adapter) {
+			delete(n.adapterStats, adapter)
+		}
+	}
+}
+
+func (n *networkCounter) mergePacketsReceivedPerSecondData(packetsReceivedPerSecondData map[string]uint64) []string {
+	adapters := make([]string, 0)
```

Comment:
Nitpicky but I'd say either use `make([]string, 0, len(packetsReceivedPerSecondData)` or just `var adapters []string`

Reply:
Good catch. Fixed in the new commit